pkg/flare/flare.go

```diff
@@ -0,0 +1,435 @@
+
+	w := zip.NewWriter(fl)
+	defer w.Close()
+
+	walker := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		ex, err := os.Executable()
+		if err != nil {
+			return err
+		}
+		exPath := filepath.Dir(ex)
+		relPath, err := filepath.Rel(exPath, path)
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		f, err := w.Create(relPath)
+		if err != nil {
+			return err
+		}
+
+		_, err = io.Copy(f, file)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	err = filepath.Walk(inputPath, walker)
+	return err
+}
+
+// HasLogFileOutput checks the outputs property under the logging property of the config
+// to make sure logs are configured to be written to file.
+func (f *Flare) HasLogFileOutput(outputs []string) bool {
+	return slices.ContainsFunc(outputs, func(e string) bool {
+		return e != "" && e != "-" && e != "="
+	})
+}
+
+// SupportedLogFormat checks that the string provided is one of the supported LogFormats
+func SupportedLogFormat(format string) bool {
+	v := LogFormat(format)
+	return v == LogFormatJSON || v == LogFormatPlainText
+}
+
+// ProcessLogFiles processes log files in inputPath. If the inputPath is a file, it processes the file.
+// If inputPath is a directory, all files in the directory will be individually processed.
+// Processed files are read line-by-line. Each line is filtered according to the date range,
+// sanitized from secrets, and is written to the file in outputPath.
+func (f *Flare) ProcessLogFiles(inputPath, outputPath string, startDate, endDate *time.Time) (retErr error) {
+	isDir, err := isDirectory(inputPath)
+	if err != nil {
+		return err
+	}
+
+	if !isDir {
+		return f.processLogFile(inputPath, outputPath, startDate, endDate)
+	} else {
+		files, err := os.ReadDir(inputPath)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			err = f.processLogFile(file.Name(), outputPath, startDate, endDate)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func extractDateFromPlainTextLine(line, logDateLayout string) (time.Time, error) {
+	idxs := plainTextLogDateRegex.FindIndex([]byte(line))
+	if idxs == nil {
+		return time.Time{}, ErrDateNotFound
+	}
+
+	strDate := line[idxs[0]+1 : idxs[1]-1]
+	date, err := time.Parse(logDateLayout, strDate)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return date, nil
+}
+
+func extractDateFromJSONLine(line, logDateLayout string) (time.Time, error) {
+	logPropsMap := make(map[string](interface{}))
+	err := json.Unmarshal([]byte(line), &logPropsMap)
+	if err != nil {
+		return time.Time{}, err
+	}
+	lineDateStr, ok := logPropsMap["time"].(string)
+	if !ok {
+		return time.Time{}, ErrExtractDateFromJSONLogLine
+	}
+	lineDate, err := time.Parse(logDateLayout, lineDateStr)
+	if err != nil {
+		return time.Time{}, err
+	}
```

Comment:
Up to managing timezones correctly (which I am not sure that this code does), you can get the same result by letting json.Unmarshal ignore unused fields as it does by default and just parsing into a struct that has a single field "Time".  Here's [an example](https://go.dev/play/p/QmyD70Pm0MG).

Reply:
Nice! Done.